### ISSUE TITLE
Data pipeline: data.as_numpy_iterator error

### ISSUE THREAD

**Issue by Noorallah1311** [nicknochnack/LipNet#9]
<img width="1130" alt="Screenshot 2023-07-20 at 3 43 55 PM" src="https://github.com/nicknochnack/LipNet/assets/134913135/047ed60a-9363-462c-9401-620baee5f082">
Hi, did anybody find the solution for this error? I'm running the code on Google Colab, Tensorflow 2.12.0
Any help is much appreciated.

**Comment by abhishek-sameriya** [comment]
if u find the solution please tell me i m facing this problem too

**Comment by Ameya22Joshi** [comment]
For the people who are using colab, 
- I used tensorflow version 2.10.1
- In load_data(), use file_name = path.split('/')[-1].split('.')[0]
- For mappable_function, use this code: 
from typing import Tuple
def mappable_function(path:str) ->Tuple[tf.Tensor, tf.Tensor]:
    result = tf.py_function(load_data, [path], (tf.float32, tf.int64))
    return result
(The tf.py_function call returns a tuple of two tensors (tf.float32 and tf.int64), not a list of strings.)

**Comment by kido321** [comment]
@Ameya22Joshi Can you make it more clear on how to fix this issue?

**Comment by SushovanHaldar** [comment]
@Ameya22Joshi  were you able to fix this issue? If yes can you please help us?

**Comment by Ameya22Joshi** [comment]
> @Ameya22Joshi Can you make it more clear on how to fix this issue?

--> The return type of tensors is by default in form of tuple, but previously it was in the form of a list. Hence, we have to make it in tuple form. (def mappable_function(path:str) ->Tuple[tf.Tensor, tf.Tensor]:)

**Comment by Ameya22Joshi** [comment]
> <img alt="Screenshot 2023-07-20 at 3 43 55 PM" width="1130" src="https://user-images.githubusercontent.com/134913135/254797613-047ed60a-9363-462c-9401-620baee5f082.png"> Hi, did anybody find the solution for this error? I'm running the code on Google Colab, Tensorflow 2.12.0 Any help is much appreciated.

The problem arises because there is a hidden file .ipynb_checkpoints in the data. To delete this file:

-  Use the following command after extracting the data from zip file --> rm -rf `find -type d -name .ipynb_checkpoints`

**Comment by Ameya22Joshi** [comment]
> @Ameya22Joshi were you able to fix this issue? If yes can you please help us?

Yes I was able to fix the issue. The solutions related to [data.as_numpy_iterator error] and [data/alignments/s1/.align] file not found error] are mentioned in the comments/replies.

**Comment by Sheeevani** [comment]
**install specific version of tensorflow**
pip install tensorflow==2.10.1

**Make changes in load_data function, update this line as**
file_name = path.split('/')[-1].split('.')[0]

**Change '\\' to '/' in below variables** 
test_path = './data/s1/bbal6n.mpg'
tf.convert_to_tensor(test_path).numpy().decode('utf-8').split('/')[-1].split('.')[0]

It will work fine then without any error in nextiterator :)

**Comment by sphynxlee** [comment]
> **install specific version of tensorflow** pip install tensorflow==2.10.1
> 
> **Make changes in load_data function, update this line as** file_name = path.split('/')[-1].split('.')[0]
> 
> **Change '\' to '/' in below variables** test_path = './data/s1/bbal6n.mpg' tf.convert_to_tensor(test_path).numpy().decode('utf-8').split('/')[-1].split('.')[0]
> 
> It will work fine then without any error in nextiterator :)

Thank you, works for me!
